src/fabric/fabricHelper.ts

```diff
@@ -0,0 +1,225 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import { FabricSqlDbInfo, IWorkspace, IFabricError } from "../sharedInterfaces/connectionDialog";
+import { HttpHelper } from "../http/httpHelper";
+
+export class FabricHelper {
+    static readonly fabricUriBase = vscode.Uri.parse("https://api.fabric.microsoft.com/v1/");
+    constructor() {}
+
+    public static async getFabricWorkspaces(tenantId: string): Promise<IWorkspace[]> {
+        const response = await this.fetchFromFabric<{ value: IWorkspace[] }>(
+            "workspaces",
+            `listing Fabric workspaces for tenant '${tenantId}'`,
+            tenantId,
+        );
+
+        return response.value;
+    }
+
+    public static async getFabricWorkspace(
+        workspaceId: string,
+        tenantId: string,
+    ): Promise<IWorkspace> {
+        const response = await this.fetchFromFabric<IWorkspace>(
+            `workspaces/${workspaceId}`,
+            `getting Fabric workspace '${workspaceId}'`,
+            tenantId,
+        );
+
+        return response;
```

Comment:
Out of curiosity, why did you use response.value on line 21, but not here?

Reply:
That's just the way the Fabric REST API returns data: [List Workspaces](https://learn.microsoft.com/en-us/rest/api/fabric/admin/workspaces/list-workspaces?tabs=HTTP) sticks the array under `.value`, but [GetWorkspace](https://learn.microsoft.com/en-us/rest/api/fabric/admin/workspaces/get-workspace?tabs=HTTP#examples) returns it directly.